### PR TITLE
A new Soundex filter for matching 'name' with people using Soundex

### PR DESCRIPTION
**Pull request by prculley**
A response to [request](http://gramps.1791082.n4.nabble.com/Spelling-variations-and-the-SoundEx-algorithm-td4678729.html) on Gramps developers email list.

### CODE REVIEW

File: gramps/gen/filters/rules/person/_hassoundexname.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+#
+# Gramps - a GTK+/GNOME based genealogy program
+#
+# Copyright (C) 2002-2007  Donald N. Allingham
+# Copyright (C) 2007-2008  Brian G. Matherly
```
Comment (QuLogic) [inline gramps/gen/filters/rules/person/_hassoundexname.py]: New file should get a reasonably new copyright date?

Comment (prculley) [inline gramps/gen/filters/rules/person/_hassoundexname.py]: I should probably take credit as well...

File: gramps/gen/filters/rules/person/_hassoundexname.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    allow_regex = False
+
+    def apply(self, db, person):
+        for name in [person.get_primary_name()] + person.get_alternate_names():
+            if self._match_name(name):
+                return True
+        return False
+
+    def _match_name(self, name):
+        if self.list[0] and self._soundex_match(0, name.get_first_name()):
```
Comment (QuLogic) [inline gramps/gen/filters/rules/person/_hassoundexname.py]: This is a bit repetitive; can be written as:
```python
if self.list[0]:
    if self._soundex_match(0, name.get_first_name()):
        return True
    elif self._soundex_match(0, name.get_surname()):
        return True
    ...

    for surn in name.get_surname_list():
        ...

return False
```
(and remove the `self.list[0]` check in `_match_surname`.)

Comment (QuLogic) [inline gramps/gen/filters/rules/person/_hassoundexname.py]: Actually, you always pass `param_index=0`, so why not just send `self.list[0]` directly to `_soundex_match`?

**Comment by sam-m888** [comment]
Works for me.

General filter > **Soundex match of People with the <name>** 
soundex match of people with the _name_ filter

 - [ ] Update documentation when it rolls over.

**Comment by sam-m888** [comment]
Related feature request [1768: improve soundex tool. Integrate better with gramps](https://gramps-project.org/bugs/view.php?id=1768)  which asks for:

> 2. Allow finding surnames with the same soundex code. E.g. have an option to select a surname and then do a "find surnames with same soundex code" This could be a quick view and should also be possible in the soundex tool.

**Comment by sam-m888** [comment]
Thank you

**Pull request merged**